### ISSUE TITLE
ProB2-UI java.io.IOException when SVG file is empty

### ISSUE THREAD

**Issue by leuschel** [hhu-stups/prob-issues#290]
When no SVG file is provided for VisB the
call prob2_visb_file_loaded(JsonFile,SvgFile) returns the empty string '' for SvgFile.
This leads to an exception in Java.

```
Caused by: java.io.IOException: Is a directory
        at java.base/sun.nio.ch.FileDispatcherImpl.read0(Native Method)
        at java.base/sun.nio.ch.FileDispatcherImpl.read(FileDispatcherImpl.java:48)
        at java.base/sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:330)
        at java.base/sun.nio.ch.IOUtil.read(IOUtil.java:296)
        at java.base/sun.nio.ch.IOUtil.read(IOUtil.java:273)
        at java.base/sun.nio.ch.FileChannelImpl.read(FileChannelImpl.java:229)
        at java.base/sun.nio.ch.ChannelInputStream.read(ChannelInputStream.java:65)
        at java.base/sun.nio.ch.ChannelInputStream.read(ChannelInputStream.java:107)
        at java.base/sun.nio.ch.ChannelInputStream.read(ChannelInputStream.java:101)
        at java.base/java.nio.file.Files.read(Files.java:3244)
        at java.base/java.nio.file.Files.readAllBytes(Files.java:3295)
        at de.prob2.ui.visb.VisBStage.loadSvgFile(VisBStage.java:361)
        at de.prob2.ui.visb.VisBStage.lambda$initialize$15(VisBStage.java:220)
```
An example can be found here:
prob_examples/examples/B/VisB/TrainNonLinear
Note: the example also uses the new attribute:
  "svg_box": {"width":800, "height":300},
Which should ideally also be supported by VisB. Here we need a new or an updated API call (prob2_visb_file_loaded).

**Comment by favu100** [comment]
Discuss whether to generate an empty SVG or provide the SVG content from Prolog with width and height. 
Furthermore adapt width and height to size provided in VisB file.

For now, the issue is solved with hard-coded height and width

**Comment by leuschel** [comment]
Also, ProB can now in principle deal with empty JSON files (VISB_JSON_FILE == "").

In ProB2-UI this does not work yet:
```
java.io.IOException: given json path is not a regular file: /Users/leuschel/git_root/THALES/fm-tool-evaluation/visualisation
	at de.prob2.ui.visb.VisBFileHandler.constructVisualisationFromJSON(VisBFileHandler.java:45)
	at de.prob2.ui.visb.VisBController.setupVisualisation(VisBController.java:260)
	at de.prob2.ui.visb.VisBController.lambda$initialize$1(VisBController.java:95)
	at javafx.base/com.sun.javafx.binding.ExpressionHelper$Generic.fireValueChangedEvent(ExpressionHelper.java:360)
	at javafx.base/com.sun.javafx.binding.ExpressionHelper.fireValueChangedEvent(ExpressionHelper.java:80)
	at javafx.base/javafx.beans.property.ObjectPropertyBase.fireValueChangedEvent(ObjectPropertyBase.java:106)
	at javafx.base/javafx.beans.property.ObjectPropertyBase.markInvalid(ObjectPropertyBase.java:113)
	at javafx.base/javafx.beans.property.ObjectPropertyBase.set(ObjectPropertyBase.java:147)
	at de.prob2.ui.visb.VisBController.setVisBPath(VisBController.java:122)
	at de.prob2.ui.visb.VisBStage.loadVisBFileFromMachine(VisBStage.java:302)
	at de.prob2.ui.visb.VisBStage.lambda$initialize$16(VisBStage.java:231)
	at de.prob2.ui.visb.VisBStage.lambda$initialize$18(VisBStage.java:248)
...
```

**Comment by leuschel** [comment]
An example is at prob_examples/public_examples/B/VisB/FunctionGraph.mch

**Comment by favu100** [comment]
It should now be possible to use VisB from DEFINITIONS. The height and width are still hardcoded into the SVG content

**Comment by leuschel** [comment]
I have added the predicate prob2_visb_default_svg_file_contents/1 to prob2_interface.pl.
This predicate should be used when no SVG file is provided and then returns a default SVG file with the correct height and width and possibly additional content (from VISB_SVG_CONTENT definitions).

**Comment by dgelessus** [comment]
This was implemented in hhu-stups/prob2_ui@f60932dfefc944d39f96aeb499b8ec6593618966 - ProB 2 UI now uses the above command to get the SVG code when no file path is set.
